### PR TITLE
in progress

### PR DESCRIPTION
**Pull request by gregemax**
rounding up with [FRontend] Vault Stake page [mobile] #304

### CODE REVIEW

**Comment by djeck1432** [comment]
@gregemax Please, you ran out of time, so please, finish it right away, we will merge everyting in 12 hours. Don't hesitate.
if there is no finish PR in an hour, you will be unassigned

File: frontend/src/pages/vault/stake/Stake.jsx
Comment (whateverfw) [inline frontend/src/pages/vault/stake/Stake.jsx:1]: Also on desktop card seems not to be visible as in Figma, not sure if that was done by your changes but  could you also fix that 
image

**Review by whateverfw — CHANGES_REQUESTED** [review]

**Comment by gregemax** [comment]
Noted. I'll be on it

**Comment by djeck1432** [comment]
@gregemax Please, we don't have a lot of time, fix it immediately

**Comment by whateverfw** [comment]
@gregemax 
1) everything is duplicated on a page
2) fix CI
3) starknet icon in options now broken
4) top cards broken on mobile, but should've fit screen
image

**Comment by djeck1432** [comment]
@gregemax please, fix also test cases

**Review by whateverfw — CHANGES_REQUESTED** [review]
@gregemax 
1) desktop version is now broken, fix it according to figma
__
still unresolved after last commit:
2) starknet icon in options now broken
3) top cards broken on mobile, but should've fit screen

**Comment by djeck1432** [comment]
@gregemax please, fix your issues as soon as possible

**Comment by djeck1432** [comment]
@gregemax as far as vault will be in next release, you will have more time, but please, remember about 12hours update, if there is any update during 12 hours from your side, you will be unassigned

**Comment by whateverfw** [comment]
image
1) still broken on <768px devices
2) also page bg don't correspond to figma
3) main content doesn't have borders on desktop, but has on mobile, please fix as well
image


@gregemax

**Review by whateverfw — CHANGES_REQUESTED** [review]
@gregemax ⏫

**Review by whateverfw — CHANGES_REQUESTED** [review]
desktop version is broken now, it should be accroding to figma...

**Review by whateverfw — CHANGES_REQUESTED** [review]
1) You can clearly see that 2 cards fit 390 screen in figma but not in the app....
image

2) Also starknet icon is not displayed on mobile (inside a top-card)

3) I can't open a dropdown, it should be opened whenever I click any area if a selector

**Comment by whateverfw** [comment]
@gregemax resolve conflicts with main, and make sure it doesn't break withdraw page as well

**Comment by whateverfw** [comment]
@gregemax please fix failed build and comments above.

**Review by whateverfw — CHANGES_REQUESTED** [review]
Withdraw page completely dissapeared on mobile after the merge with main, probably you merged it wrong.
image

**Review by whateverfw — CHANGES_REQUESTED** [review]
image
stake page is also not completely working well.

**Comment by whateverfw** [comment]
@gregemax any updates?

**Comment by gregemax** [comment]
working on it

**Comment by djeck1432** [comment]
@gregemax any updates?

File: frontend/src/pages/vault/stake/Stake.jsx [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
+import BalanceCards from 'components/BalanceCards';
+import { Image } from 'react-bootstrap';
 
 function Stake() {
   const [selectedNetwork, setSelectedNetwork] = useState('Starknet');
   const [amount, setAmount] = useState('0');
 
-  const networks = [{ name: 'Starknet', image: STRK }];
+  const [balances, setBalances] = useState([
+    { icon: <Image src={STRKIMG} />, title: 'STRK', balance: '0.046731' },
```
Comment (whateverfw) [inline frontend/src/pages/vault/stake/Stake.jsx:19]: use it just like other components

File: frontend/src/pages/vault/stake/stake.css [hunk 2]
```diff
@@ -144,3 +173,282 @@
 .stake-button {
   width: 100%;
 }
+
+
+/* here */
```
Comment (whateverfw) [inline frontend/src/pages/vault/stake/stake.css:178]: remove comment

File: frontend/src/pages/vault/stake/stake.css [hunk 3]
```diff
@@ -144,3 +173,282 @@
...
+  }
+  
+
+}
+
+/* Desktop styles */
+@media (min-width: 1025px) {
+  /* .desktop-layout {
+    display: block;
+  } */
```
Comment (whateverfw) [inline frontend/src/pages/vault/stake/stake.css:274]: remove comment

File: frontend/src/pages/vault/stake/stake.css [hunk 4]
```diff
@@ -144,3 +173,282 @@
...
+    border: 1px solid #36294E;
+    background-color: #120721;
+    border-radius: 16px;
+    margin-top: 10px;
+    min-height: 305px;
+    width: 342px;
+    margin-left: auto;
+    margin-right: auto;
+  }
+ */
```
Comment (whateverfw) [inline frontend/src/pages/vault/stake/stake.css:442]: remove commented code

File: frontend/src/pages/vault/stake/stake.css
Comment (whateverfw) [inline frontend/src/pages/vault/stake/stake.css:1]: page background still do not correpond to figma.

**Review by whateverfw — CHANGES_REQUESTED** [review]
1)

File: frontend/src/pages/vault/stake/Stake.jsx [hunk 5]
```diff
@@ -24,25 +35,22 @@ function Stake() {
...
-              <MetricCard title="STRK Balance" value="0.046731" icon={STRK} />
-              <MetricCard title="APY Balance" value="0.046731" icon={USDC} />
-            </div>
-          </div>
-
           <h1 className="stake-title">Please submit your leverage details</h1>
-
           <div className="main-card">
             <div className="network-selector-container">
               <div className="network-selector">
```
[the remark this answers is not in the document]
Comment (gregemax) [reply frontend/src/pages/vault/stake/Stake.jsx:69]: Hello @whateverfw  
The network selector issue was present when I pulled the code from the main branch, so I believe it's not from my changes. I commented out the previous and put in this new change  , 
// const handleChange = (e) => {
  //   setSelectedNetwork(e.target.value);
  // };
  const handleChange = (network) => {
    setSelectedNetwork(network.name);
  };
but I’m not sure if that resolves the problem.
and again why the network selector 
Hover Removed: Mobile devices don't support hover, so  I will use click event.

File: frontend/src/pages/vault/stake/stake.css
Comment (whateverfw) [reply frontend/src/pages/vault/stake/stake.css:1]: you can clearly see, even in colors, that bg is still not the same as in figma

[the remark this answers is not in the document]
Comment (whateverfw) [reply frontend/src/pages/vault/stake/stake.css:1]: image
mobile is broken and don't look like in figma

File: frontend/src/pages/vault/stake/Stake.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (whateverfw) [reply frontend/src/pages/vault/stake/Stake.jsx:69]: I understand, but obviously when you merge something in your branch, you should do it properly and verify that everything works, correctly. So yeah, now it works no mobile, thanks, but remove commented code, it is redundant to leave.

File: frontend/src/pages/vault/stake/Stake.jsx
Comment (whateverfw) [inline frontend/src/pages/vault/stake/Stake.jsx:1]: Also, please format your file, thanks

**Review by whateverfw — CHANGES_REQUESTED** [review]

File: frontend/src/pages/vault/stake/Stake.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (whateverfw) [reply frontend/src/pages/vault/stake/Stake.jsx:70]: @gregemax

File: frontend/src/pages/vault/stake/stake.css
[the remark this answers is not in the document]
Comment (whateverfw) [reply frontend/src/pages/vault/stake/stake.css:1]: @gregemax

**Review by whateverfw — CHANGES_REQUESTED** [review]
image

Padding from right and left is not equal, I can clearly see it..

Top cards sizes do not correspond to figma.

**Review by whateverfw — CHANGES_REQUESTED** [review]
broken 
image

**Review by whateverfw — CHANGES_REQUESTED** [review]
broken
image

**Comment by whateverfw** [comment]
@gregemax can you please test your changes before your commits and tag me only when you verify by yourself that bugs are fixed?

everything is still broken what was mentioned above, the only thing is mobile paddings and sizes had been improved, but now another thing is broken like that.
image

**Comment by djeck1432** [comment]
@gregemax are you sure you can manage it,I see every next commit you have new issue?

**Comment by gregemax** [comment]
Hello  @whateverfw  are you available for a Google meet  ? please  
Thanks

**Comment by whateverfw** [comment]
> Hello @whateverfw are you available for a Google meet ? please Thanks

@gregemax You can write your questions here.

**Review by djeck1432 — APPROVED** [review]
Partially done work,

**Pull request merged**